include/CLI/Optional.hpp

```diff
@@ -15,19 +15,25 @@
 #if defined(CLI11_CPP17) && __has_include(<optional>) && \
      !defined(CLI11_STD_OPTIONAL)
 #define CLI11_STD_OPTIONAL 1
+#elif !defined(CLI11_STD_OPTIONAL)
+#define CLI11_STD_OPTIONAL 0
```

Comment:
I was getting a bunch of warnings about undefined symbols when this is used in #if XXXX statements

Reply:
It was supposed to be checked in `ifdef`, but it was checked in an `if` just below, so this is probably best. The main user interaction with these is optionally defining 1 or 0 before including, and if they do define 0, it would have been inconsistent before this patch, so I like it.